core/src/main/java/io/opencensus/tags/TagContextBinarySerializer.java

```diff
@@ -29,25 +27,24 @@
   /**
    * Serializes the {@code TagContext} into the on-the-wire representation.
    *
-   * <p>This method should be the inverse of {@link #deserialize}.
+   * <p>This method should be the inverse of {@link #fromByteArray}.
    *
    * @param tags the {@code TagContext} to serialize.
-   * @param output the {@link OutputStream} to write the serialized tags to.
-   * @throws IOException if there is an {@code IOException} while writing to {@code output}.
+   * @return the on-the-wire representation of a {@code TagContext}.
    */
-  public abstract void serialize(TagContext tags, OutputStream output) throws IOException;
+  public abstract byte[] toByteArray(TagContext tags);
 
   /**
    * Creates a {@code TagContext} from the given on-the-wire encoded representation.
    *
-   * <p>This method should be the inverse of {@link #serialize}.
+   * <p>This method should be the inverse of {@link #toByteArray}.
    *
-   * @param input on-the-wire representation of a {@code TagContext}.
-   * @return a {@code TagContext} deserialized from {@code input}.
-   * @throws IOException if there is a parse error or an {@code IOException} while reading from
-   *     {@code input}.
+   * @param bytes on-the-wire representation of a {@code TagContext}.
+   * @return a {@code TagContext} deserialized from {@code bytes}.
+   * @throws IOException if there is a parse error.
```

Comment:
I agree, there is probably a better checked exception - possibly ParseException or DataFormatException but we can look into it more.

Reply:
Completely agreed. I think we can change the Exception later but it has to be checked.

Reply:
why does it need to be a checked exception?

Reply:
inevitably some impl will throw a kind of unchecked exception, so we have to catch RuntimeException anyway. Languages like kotlin prefer unchecked. In the case of not doing I/O, I'd personally prefer a convention of using IllegalArgumentException (or Malformed subtype of RuntimeException), as it avoids bikeshedding about checked vs unchecked.